### DISCUSSION TITLE
Frigate bundled go2rtc doesn't persist configuration changes across LXC restarts

### DISCUSSION THREAD

**Discussion by willgryan** [tteck/Proxmox#3375] [Help Needed] [answered]
### Please verify that you have read and understood the guidelines.

yes

### A clear and concise description of the issue.

After making changes to the go2rtc configuration using the go2rtc config file editor @ ip_address:1984/editor.html, they don't persist after the Frigate LXC container restarts. 

go2rtc reports the configuration path is /dev/shm/go2rtc.yaml, which seems to imply it is being placed by default in the wrong location

### What settings are you currently utilizing?

Default Settings

### Which Linux distribution are you employing?

Debian 11

### Please provide detailed steps to reproduce the issue.

1. Install Frigate LXC
2. Add stream to go2rtc using the go2rtc browser-based config file editor, then click "Save & Restart" on webpage
3. Verify stream can be viewed in go2rtc
5. Restart Frigate LXC
6. go2rtc configuration is now reset to default values

**Comment by tteck** [comment]
I don't utilize Frigate, so we'll have to wait for someone who does to chime in.

**Comment by willgryan** [comment, marked as answer]
I have learned that if you use go2rtc bundled with Frigate, you must configure go2rtc through Frigate ([see here](https://docs.frigate.video/guides/configuring_go2rtc)) rather than through go2rtc itself. 

Frigate generates a configuration file for go2rtc from it's own configuration when it runs. [See here](https://github.com/blakeblackshear/frigate/blob/dev/docker/main/rootfs/etc/s6-overlay/s6-rc.d/go2rtc/run#L48).

**Reply by pieterdevriesch** [reply]
I do use Frigate and just reinstalled it using the latest lxc script. Now I am noticing that it is not in fact copying over the confuration from the Frigate config. However when I configure go2rtc manually using the /dev/shm/go2rtc.yaml it seems to stick (and work). So for me it's working but this is not how it's supposed to work. If I can assist with any further information let me now.

**Reply by willgryan** [reply]
How are you configuring go2rtc in the frigate config? Here is an example of how I have done it with Frigate 0.13.2 and go2rtc 1.9.4.

```
go2rtc:
  streams:
    test:
      - rtsp://admin:password@192.168.1.100:10554/tcp/av0_0
```

When you restart the container, do your manual changes to /dev/shm/go2rtc.yaml persist? Frigate is supposed to delete and recreate this when it starts.

**Reply by pieterdevriesch** [reply]
Yes that's exactly how I have it, it does persist after restart, so it seems like the whole delete and copy operation is not happening.

**Comment by gideonyong** [comment]
I'm running into the same issue. If i delete /dev/shm/go2rtc.yaml, then the file is recreated and is now correct.  

Any changes to the go2rtc part of the config.yml file are not being copied over to go2rtc.yaml file.  the only way for this to work is to delete the go2rtc.yaml file and then it will correctly re-create the updated config from the config.yml

**Reply by grbasta** [reply]
Question, do you delete the entire path or just the go2rtc.yaml file? By deleting the file, the default one is reloaded anyway and so I have to reload the configuration every time

**Reply by gideonyong** [reply]
I just deleted the yaml file.  after i rebooted, a new yaml file was automatically created in the same folder.

**Comment by phalox** [comment]
Same issue here, go2rtc config file isn't being updated. Only after a restart the file is re-created. some permission issues?

**Reply by phalox** [reply]
But if I understand correctly, that's by design: Only when the docker containers are rerun, the code would execute.... ? What's the clean process to do so then?

**Reply by grbasta** [reply]
It's a real shame because every time you restart the container, go2rtc.yaml loses the configuration, and then at least I was unable to reconfigure the operation with google coral and was forced to use openvino.
